### PR TITLE
src: Allow the `buffer` command to jump to line/column

### PR DESCRIPTION
**Pull request by lenormf**
Hi,

When I have several buffers opened in different directories and want to jump to a known location in one of them, using `edit` is too tedious if the buffer is far away from the CWD.

HTH.

### CODE REVIEW

**Comment by danr** [comment]
This can also be achieved by using `select` after `edit`.

**Comment by lenormf** [comment]
I can also use `<n>g` but that's still one more command I don't want to use if possible.

**Comment by Delapouite** [comment]
On a side note, there's no easy way to goto a specific column.

File: src/commands.cc [hunk 1]
```diff
@@ -514,11 +514,14 @@ const CommandDesc write_all_quit_cmd = {
     }
 };
 
+ParameterDesc buffer_params{
+    {}, ParameterDesc::Flags::None, 1, 3
```
Comment (mawww) [inline src/commands.cc]: This is not reused, it can just be written in line in the CommandDesc

File: src/commands.cc [hunk 2]
```diff
@@ -529,6 +532,18 @@ const CommandDesc buffer_cmd = {
         {
             context.push_jump();
             context.change_buffer(buffer);
+
+            const size_t param_count = parser.positional_count();
+            if (param_count > 1 and not parser[1].empty())
+            {
+                int line = std::max(0, str_to_int(parser[1]) - 1);
+                int column = param_count > 2 and not parser[2].empty() ?
```
Comment (mawww) [inline src/commands.cc]: const LineCount/ByteCount instead of int would be nice here

File: src/commands.cc [hunk 3]
```diff
@@ -529,6 +532,18 @@ const CommandDesc buffer_cmd = {
         {
             context.push_jump();
             context.change_buffer(buffer);
+
+            const size_t param_count = parser.positional_count();
+            if (param_count > 1 and not parser[1].empty())
```
Comment (mawww) [inline src/commands.cc]: I am not sure we need to test for empty, str_to_int will return 0 in those cases. (Maybe it should fail with "not a number", in any case we dont gain much by testing here).

**Comment by lenormf** [comment]
I think all the comments have been addressed.

**Comment by lenormf** [comment]
Bump!

**Comment by Delapouite** [comment]
Should we also document these new optional params in https://github.com/mawww/kakoune/blob/master/doc/pages/commands.asciidoc ?

**Comment by eraserhd** [comment]
I'm not clear on what the use case is.  I assume you want to use the `:buffer` command to auto-complete the buffer name at the prompt (which is why this helps when it is far away from the CWD), but then you want to type in the line and column manually?

**Comment by mawww** [comment]
Yeah, it is still unclear to me what the use case for this is.

**Comment by lenormf** [comment]
When I `grep` for something in another shell and get a line number, using `:buffer` and its fuzzy matching completion engine is more convenient than using `:edit` (which returns all files on disk as candidates).

**Comment by eraserhd** [comment]
I think you are saying `:buffer`'s autocomplete is better.  Have you looked at `:find` in Maxime's kakrc?

**Comment by mawww** [comment]
Thinking of it, for consistency with edit it could make sense to support this, but at the same time it means we lose the selections that were in whatever window we end-up recycling.
